SeaBattle/BaseShip.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+using System.Drawing;
+using System.Reflection;
+using System.Xml.Linq;
+using System.Data.SqlTypes;
+
+namespace SeaBattle
+{
+    public class BaseShip
+    { 
+        public List<Deck> Decks { get; set; }
+        int Durability { get; set; }
+        int MoveSpeed { get; set; }
+        int MoveDirectionX { get; set; }
+        int MoveDirectionY { get; set; }
+        string Name { get; set; }
+        int Speed { get; set; }
+        
+        public BaseShip(string Name, List<Point> points) {
+            var Decks = new List<Deck>();
+
+            for (int i = 0; i < points.Count - 1; i++)
+            {
+                Deck a = new Deck();
+                a.Location = points[i];
+                Decks.Add(a);
+            }
+        }
+        public override string ToString()
+        {
+            string output = "Name: " + Name + ", ";
+            output += "Type: " + this.GetType() + ", ";
+            output += "Durability: " + Durability + ", ";
+            output += "Speed: " + Speed + ", ";
+            output += "Decks:" + Decks.Count + ", ";
+            output += "Coordinates: ";
+            foreach (var deck in Decks)
+            {
+                output +=  deck.Location.X + ":" + deck.Location.Y + ", ";
+            }
+            return output;
+        }
+        public static bool operator== (BaseShip s1, BaseShip s2)
+        {
+            return ((s1.Decks.Count == s2.Decks.Count) && (s1.GetType() == s2.GetType()));
+        }
+        public static bool operator!= (BaseShip s1, BaseShip s2)
+        {  
+            return !((s1.Decks.Count == s2.Decks.Count) && (s1.GetType() == s2.GetType()));
+
+        }
+
+        // This function adds s1(BaseShip) to the Sea
+        // 1. + Checking if Point within Sea borders
+        // 2. + Checking if Point is available 
+        
+        public void AddShip(BaseShip s1, Sea PlaceSea)
+        {
+
+            foreach (var d in s1.Decks)
+            {
+                if (d.Location.X > PlaceSea.SeaWidth || d.Location.Y > PlaceSea.SeaHeight || d.Location.X < 0 || d.Location.Y < 0)
+

```

Comment:
empty space, a lot of them everywhere. please pay attention to this and remove all of them.

Reply:
trying to fix it every commit..